### PR TITLE
tgl: iccmax

### PR DESCRIPTION
**Pull request by abonislawski**
This patch will update our clock change procedure to align
with hardware recommended flow for TGL platforms

### CODE REVIEW

**Comment by abonislawski** [comment]
SOFCI TEST

File: src/arch/xtensa/lib/cpu.c [hunk 1]
```diff
@@ -37,6 +37,7 @@ extern struct core_context *core_ctx_ptr[PLATFORM_CORE_COUNT];
 extern struct xtos_core_data *core_data_ptr[PLATFORM_CORE_COUNT];
 
 static uint32_t active_cores_mask;
+int dsp_sref[PLATFORM_CORE_COUNT];	/**< simple ref counter */
```
Comment (kv2019i) [inline src/arch/xtensa/lib/cpu.c:40]: @abonislawski Should this be static?

File: src/platform/intel/cavs/lib/clk.c [hunk 2]
```diff
@@ -48,6 +53,10 @@ static inline void select_cpu_clock(int freq_idx, bool release_unused)
 		io_reg_write(SHIM_BASE + SHIM_CLKCTL,
 			     (io_reg_read(SHIM_BASE + SHIM_CLKCTL) &
 			      ~SHIM_CLKCTL_OSC_REQUEST_MASK) | enc);
+#if CONFIG_TIGERLAKE
+		if (freq_idx == CPU_LPRO_FREQ_IDX)
+			cpu_disable_core(PLATFORM_CORE_COUNT - 1);
+#endif
```
Comment (kv2019i) [inline src/platform/intel/cavs/lib/clk.c:59]: I guess this is on purpose, but looks a bit hard to follow. Wouldn't the correct logic be to disable the count-1 cpu core if any other freq than CPU_HPRO is requested (and not just CPU_LPRO_FREQ_IDX)?

File: src/arch/xtensa/lib/cpu.c (hunk 1 above)
Comment (abonislawski) [reply src/arch/xtensa/lib/cpu.c]: yes, it should, fixed

File: src/platform/intel/cavs/lib/clk.c (hunk 2 above)
Comment (abonislawski) [reply src/platform/intel/cavs/lib/clk.c:59]: well, it doesnt matter because we have only HPRO and LPRO on TGL :)
But I can update with your approach
